7_react/react1/week3/vite-project/src/components/addNewToDo.jsx

```diff
@@ -0,0 +1,168 @@
+import { useState, useEffect } from "react";
+import ListItemWithBorder from "./listItemWithBorder.jsx";
+
+const API_URL="https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+function AddNewToDo() {
+
+  const [todos, setTodos] = useState([]);
+  const [inputNewTodo, setInputNewTodo] = useState("");
+  const [inputDeadline, setInputDeadline] = useState("");
+  const [editModeId, setEditModeId] = useState(null);
+  const [editDescription, setEditDescription] = useState("");
+
+  useEffect(() => {
+    fetch(
+      API_URL
+      )
+      .then((response) => response.json())
+      .then((data) => setTodos(data))
+      .catch((error) => (error.message));
+  }, []);
+
+
+  const handleAddTodo = (e) => {
+    e.preventDefault();
+    if (inputNewTodo.trim() !== "" && inputDeadline !== "") {
+      setTodos([
+        ...todos,
+        {
+          id: Math.random(),
+          description: inputNewTodo.trim(),
+          deadline: inputDeadline,
+          checked: false,
+        },
+      ]);
+      setInputNewTodo("");
+      setInputDeadline("");
+    } else {
+      alert("Please enter a task and deadline");
+    }
+  };
+
+  const handleChecked = (id) => {
+    const updatedTodos = todos.map((todo) =>
+      todo.id === id ? { ...todo, checked: !todo.checked } : todo
+    );
+    setTodos(updatedTodos);
+  };
+
+  const handleDeleteTodo = (id) => {
+    setTodos(todos.filter((todo) => todo.id !== id));
+  };
+
+  const handleEditClick = (id, description) => {
+    setEditModeId(id);
```

Comment:
This choice of state and component layout has some effects that may or may not be desirable. The most obvious is that you can only edit one todo at a time; that might be fine, but could also be unnecessarily restricting. As alluded to further up, it might be a lot easier to do if you split the component up a bit. If you would like to support this, can you think of how to do it? 🙂

I did discover a bug: if you edit a todo and delete it before pressing update, you can no longer edit any todos. Can you see why? And can you find a solution that both fixes this issue and perhaps makes it easier to support multi-editing? 😄

Reply:
Regarding this part: 'If you would like to support this, can you think of how to do it? ', could you please give me a hint, thank you 😊

'And can you find a solution that both fixes this issue and perhaps makes it easier to support multi-editing?' I will try that 😊